### PR TITLE
app-crypt/tpm-tools: fix build with openssl-1.1

### PR DESCRIPTION
**Pull request by ConiKost**
Closes: https://bugs.gentoo.org/672756
Package-Manager: Portage-2.3.52, Repoman-2.3.12
Signed-off-by: Conrad Kostecki <conrad@kostecki.com>

### CODE REVIEW

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
++	const BIGNUM *bd;
++	const BIGNUM *bp;
++	const BIGNUM *bq;
++	const BIGNUM *bdmp1;
++	const BIGNUM *bdmq1;
++	const BIGNUM *biqmp;
++
++	RSA_get0_key( a_pRsa, &bn, &be, &bd);
++	RSA_get0_factors( a_pRsa, &bp, &bq);
++	RSA_get0_crt_params( a_pRsa, &bdmp1, &bdmq1, &biqmp );
```
Comment (alonbl) [inline app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: as far as I understand these are not available in openssl-1.0 nor libressl, if to use them need conditional or emulation.

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+@@ -691,8 +691,13 @@ createRsaPubKeyObject( RSA
+ 
+ 	int  rc = -1;
+ 
+-	int  nLen = BN_num_bytes( a_pRsa->n );
+-	int  eLen = BN_num_bytes( a_pRsa->e );
++	const BIGNUM *bn;
++	const BIGNUM *be;
++
++	RSA_get0_key( a_pRsa, &bn, &be, NULL );
```
Comment (alonbl) [inline app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: as far as I understand this is not available in openssl-1.0 nor libressl, if to use them need conditional or emulation.

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
++
++	int  nLen = BN_num_bytes( bn );
```
Comment (alonbl) [inline app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: declaring variables post statement is not standard in all compilers

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch [hunk 4]
```diff
@@ -0,0 +1,106 @@
+Author: Philipp Kern <pkern@debian.org>
+Subject: Fix openssl1.1 support in data_mgmt
+Date: Tue, 31 Jan 2017 22:40:10 +0100
+
+
```
Comment (alonbl) [inline app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: please add URL to open bug at upstream where patch is attached so we can followup

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch (hunk 1 above)
Comment (ConiKost) [reply app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: Ok, what do you want me to do? Drop that line? Sorry for asking, but I don't have much knowledge about openssl in terms of code. Applies for the other 3 comments too.

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch (hunk 4 above)
Comment (ConiKost) [reply app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: Where is none. Will open one.

File: app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch (hunk 1 above)
Comment (alonbl) [reply app-crypt/tpm-tools/files/tpm-tools-1.3.9.1-openssl-1.1.patch]: ok, thanks, I will try to workout something

**Comment by ConiKost** [comment]
Thank you! :)

**Pull request closed without merging**